### ISSUE TITLE
Segfault in transport

### ISSUE THREAD

**Issue by bhourahine** [dftbplus/dftbplus#820]
**Describe the bug**

Segfault 

**To Reproduce**

Empty Analysis {} block for a two terminal calculation, occurs with current main repository (and probably historical versions). Occurred for solver = transportonly {}.

**Example input**

```
Geometry = GenFormat {
  <<< 'cnt_4pl.gen'
}

Transport {
    Device {
      AtomRange = 1 80
      FirstLayerAtoms = 1  41
    }
    Contact {
      Id = "source"
      AtomRange = 81 120
      FermiLevel [eV] = -4.370889992000
      potential [eV]  = 3
    }
    Contact {
      Id = "drain"
      AtomRange = 121 160
      FermiLevel [eV] = -4.370889992000
      potential = 0.0
    }
}

Hamiltonian = DFTB {
  SCC = No
  MaxAngularMomentum = {
    C = "p"
  }
  SlaterKosterFiles = {
    C-C = "./C-C.skf"
  }
  Solver = TransportOnly {}
}

Analysis{
  #TunnelingAndDOS{}
}
```

**Output**

```
***  Geometry step: 0

Total Energy:                        0.0000000000 H            0.0000 eV


 Release NEGF memory:

Program received signal SIGSEGV: Segmentation fault - invalid memory reference.

Backtrace for this error:
#0  0x7f6fd276fd21 in ???
#1  0x7f6fd276eef5 in ???
#2  0x7f6fd243720f in ???
#3  0x5586c8700938 in __libnegf_MOD_destroy_matrices
```

**Comment by bhourahine** [comment]
Upstream [PR](https://github.com/libnegf/libnegf/pull/59) will close this.

**Comment by bhourahine** [comment]
Closed by #996
